internal/tasks/subscriptiontask.go

```diff
@@ -194,10 +194,8 @@ func (sm *SubscriptionTask) handleNodeBindingUpdates(ctx context.Context, newBin
 			if handler, ok := sm.pushHandlers.Load(oldKey); ok && handler != nil {
 				logger.Ctx(ctx).Infow("handler found, calling stop", "key", oldKey)
 				go func(ctx context.Context) {
```

Comment:
Can we avoid invoking a goroutine here?

Reply:
handler.Start is a waiting call..
https://github.com/razorpay/metro/blob/master/internal/stream/stream.go#L94
It will not exit until the stream exits.